File: db/entity/Student.java
// campuslk-api/src/main/java/lk/campuslk/db/entity/Student.java
package lk.campuslk.db.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "students")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Student {
    @Id
    @Column(length = 36)
    private String id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(name = "password_hash", nullable = false)
    private String passwordHash;

    private String fieldOfStudy;
    private String interests;
    private String profilePicture;
    private String bio;
    private String portfolioUrl;
    private String notificationPreferences;

    @Column(nullable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Instant updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "university_id")
    private University university;
}

File: campuslk_app/lib/models/student.dart
class Student {
  final String id;
  final String name;
  final String email;
  final String? bio;
  final String? portfolioUrl;
  final String? fieldOfStudy;
  final String? interests;

  Student({
    required this.id,
    required this.name,
    required this.email,
    this.bio,
    this.portfolioUrl,
    this.fieldOfStudy,
    this.interests,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      bio: json['bio'],
      portfolioUrl: json['portfolioUrl'],
      fieldOfStudy: json['fieldOfStudy'],
      interests: json['interests'],
    );
  }
}
